ogre2/src/media/materials/programs/heat_signature_fs.glsl

```diff
@@ -46,5 +46,8 @@ float mapNormalized(float num)
 void main()
 {
   float heat = texture(RT, inPs.uv0.xy).x;
-  fragColor = vec4(mapNormalized(heat), 0, 0, 1.0);
+
+  // set g, b, a to 0. This will be used by thermal_camera_fs.glsl to determine
+  // if a particular fragment is a heat source or not
+  fragColor = vec4(mapNormalized(heat), 0, 0, 0.0);
```

Comment:
Based on my understanding, objects with a heat signature use `heat_signature_fs.glsl` instead of `thermal_camera_fs.glsl`: https://github.com/ignitionrobotics/ign-rendering/blob/c6ee8b9a772db37a27edab361a176ecee80c417e/ogre2/src/Ogre2ThermalCamera.cc#L172-L190

So, if my understanding is correct, can you help me understand how `thermal_camera_fs.glsl` will determine if the output from `heat_signature_fs.glsl` is a heat source or not? I'm having a hard time understanding how these two shader files share information with one another (they seem independent to me).

Reply:
It was a misunderstanding on my end (I didn't realize that one shader is called earlier in the pipeline than the other, but it makes sense when you see how one is called in pre render target update, and the other in post), this looks good to me :+1: